### PR TITLE
adding group by

### PR DESCRIPTION
**Pull request by rafiki270**
@tanner0101 I am not sure how to add the custom group by (to allow for `CONCAT` and other special computed stuff like `YEAR` etc ...) ... would it be fine to just allow string input?

Also, how do you handle Package file in PR's when the target hasn't been released yet?

### CODE REVIEW

**Comment by rafiki270** [comment]
1: can you make the PR into the `nio` branch? that will be merged in a few days
so let's avoid conflicts and just put it directly to there
2: we need to hide the `public enum QueryGroupBy `
make this as a struct with a static func `field` instead
you can use an internal enum as storage
but if we ever want to add a new type there, we need to make sure it's not an enum or we are stuck with it
https://github.com/vapor/fluent/blob/nio/Sources/Fluent/Query/Filter/QueryFilter.swift#L26-L57
take a look at how it's achieved there ^
and here: https://github.com/vapor/fluent/blob/nio/Sources/Fluent/Query/Filter/QueryFilter.swift#L61
that's how you make something that feels like an enum, but is a struct

**Comment by calebkleveter** [comment]
Does it follow the Swift API guidelines better to call it `group(by:)` or `groupBy(_:)`? I honestly like the way the first one looks better.

**Comment by rafiki270** [comment]
doing a NIO PR instead

**Comment by rafiki270** [comment]
@tanner0101 @calebkleveter all amended

**Comment by rafiki270** [comment]
@tanner0101 done

File: Sources/Fluent/Query/QueryGroupBy.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import CodableKit
+
+/// GROUP BY statement for QueryBuilder
+public struct QueryGroupBy {
+    public enum Storage {
```
Comment (tanner0101) [inline Sources/Fluent/Query/QueryGroupBy.swift]: this enum needs to be private, as well as the storage property

**Comment by rafiki270** [comment]
@tanner0101 done

**Pull request merged**